test/integration/auth.js

```diff
@@ -33,23 +33,75 @@ function test(utils) {
 
   var newUserUid = utils.generateRandomString(20);
   var nonexistentUid = utils.generateRandomString(20);
+  var testPhoneNumber = '+11234567890';
+  var testPhoneNumber2 = '+16505550101';
+  var nonexistentPhoneNumber = '+18888888888';
+  var updatedEmail = utils.generateRandomString(20) + '@example.com';
+  var updatedPhone = '+16505550102';
 
   var mockUserData = {
     email: newUserUid + '@example.com',
     emailVerified: false,
+    phoneNumber: testPhoneNumber,
     password: 'password',
     displayName: 'Random User ' + newUserUid,
     photoURL: 'http://www.example.com/' + newUserUid + '/photo.png',
     disabled: false,
   };
 
+  /**
+   * Helper function that deletes the user with the specified phone number
+   * if it exists.
+   * @param {string} phoneNumber The phone number of the user to delete.
+   * @return {Promise} A promise that resolves when the user is deleted
+   *     or is found not to exist.
+   */
+  function deletePhoneNumberUser(phoneNumber) {
+    return admin.auth().getUserByPhoneNumber(phoneNumber)
+      .then(function(userRecord) {
+        return admin.auth().deleteUser(userRecord.uid);
+      })
+      .catch(function(error) {
+        // Suppress user not found error.
+        if (error.code !== 'auth/user-not-found') {
+          throw error;
+        }
+      });
+  }
+
+  /**
+   * @return {Promise} A promise that resolves when test preparations are ready.
+   */
+  function before() {
+    // Delete any existing users that could affect the test outcome.
+    var promises = [
+      deletePhoneNumberUser(testPhoneNumber),
+      deletePhoneNumberUser(testPhoneNumber2),
+      deletePhoneNumberUser(nonexistentPhoneNumber),
+      deletePhoneNumberUser(updatedPhone)
+    ];
+    return Promise.all(promises)
+      .catch(function(error) {
+        utils.logFailure('setUp()', error);
```

Comment:
Thanks for catching this mislabel. I originally had named this setUp() but then renamed it to before().